pkg/flowaggregator/flowaggregator.go

```diff
@@ -275,10 +275,27 @@ func (fa *flowAggregator) InitAggregationProcess() error {
 }
 
 func (fa *flowAggregator) Run(stopCh <-chan struct{}) {
-	go fa.collectingProcess.Start()
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func() {
+		// Waiting for this function to return on stop makes it easier to set expectations
+		// when testing. Without this, there is no guarantee that
+		// fa.collectingProcess.Start() was called by the time Run() returns.
+		// It also makes more sense to ensure that fa.collectingProcess.Stop() is always
+		// called after fa.collectingProcess.Start().
+		defer wg.Done()
+		fa.collectingProcess.Start()
```

Comment:
Could it lead to dead lock? 
* `fa.collectingProcess.Start()` is a blocking call and will return only after `fa.collectingProcess.Stop()` is called.
* `fa.collectingProcess.Stop()` will only be called after `fa.collectingProcess.Start()` returns.

Reply:
yes you're right
that must be why the e2e test are failing; let me come up with an alternative